pkg/kubernetes/kubectl.go

```diff
@@ -0,0 +1,266 @@
+	"github.com/pkg/errors"
+	"github.com/segmentio/kafka-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type kubernetesClient struct {
+}
+
+// NewkubernetesClient returns kubernetes client
+func NewkubernetesClient() KubeClient {
+	return &kubernetesClient{}
+}
+
+// CreateNamespace creates required namespace in k8s cluster
+func (kubectl kubernetesClient) CreateNamespace(ctx context.Context, namespace string) (string, error) {
+	command := []string{"create", "namespace", namespace}
+	return helm.RunCmdWithTimeout(ctx, "kubectl", command)
+}
+
+func (kubectl kubernetesClient) InstallKafka(ctx context.Context, namespace, yamlFileRepo, kafkaConfigPath string) (string, error) {
+	// kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
+	createKafka := []string{"create", "-n", namespace, "-f", fmt.Sprintf("%s/%s", yamlFileRepo, kafkaConfigPath)}
+	return helm.RunCmdWithTimeout(ctx, "kubectl", createKafka)
+}
+func (kubectl kubernetesClient) CreateConfigMap(ctx context.Context, configMapName, namespace, yamlFileRepo, kafkaConfigPath, sinkConfigPath, sourceConfigPath string) (string, error) {
+	//	createConfig := []string{"create", "configmap", "s3config", fmt.Sprintf("--from-file=%s/%s --from-file=%s/%s --from-file=%s/%s --from-literal=timeinSeconds=1800", yamlFileRepo, sinkConfigPath, yamlFileRepo, sourceConfigPath, yamlFileRepo, sinkConfigPath), "-n", namespace}
+	createConfig := []string{"create", "-n", namespace, "configmap", configMapName, fmt.Sprintf("--from-file=adobe-s3-sink.properties=%s/%s", yamlFileRepo, sinkConfigPath), fmt.Sprintf("--from-file=adobe-s3-source.properties=%s/%s", yamlFileRepo, sourceConfigPath), fmt.Sprintf("--from-file=adobe-kafkaConfiguration.properties=%s/%s", yamlFileRepo, kafkaConfigPath), "--from-literal=timeinSeconds=1800"}
+	return helm.RunCmdWithTimeout(ctx, "kubectl", createConfig)
+}
+
+func (kubectl kubernetesClient) Ping(ctx context.Context, namespace string) (string, error) {
+	// kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
+	pingKafka := []string{"run", "-n", namespace, "kafka-ping", "-ti", "--rm=true", "--image=strimzi/kafka:0.20.0-kafka-2.6.0", "--restart=Never", "--", "bin/kafka-topics.sh", "--list", "--bootstrap-server=my-cluster-kafka-external-bootstrap:9094"}
+	log.Print(pingKafka)
+	return helm.RunCmdWithTimeout(ctx, "kubectl", pingKafka)
+}
+func (kubectl kubernetesClient) Insert(ctx context.Context, topic, namespace string) error {
+	err := produce(ctx, topic, namespace)
+	return err
+}
+func (kubectl kubernetesClient) Count(ctx context.Context, topic, namespace string) (int, error) {
+	count, err := consume(ctx, topic, namespace)
+	return count, err
+}
+
+// to produce messages to kafka
+func produce(ctx context.Context, topic string, namespace string) error {
```

Comment:
The Kafka-specific functions should be in pkg/app/kafka. Please refactor kubernetes package to have only client related methods